iothub/service/src/Query/QueryClient.cs

```diff
@@ -104,45 +102,31 @@ public virtual async Task<QueryResponse<T>> CreateAsync<T>(string query, QueryOp
 
             try
             {
-                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(
-                    HttpMethod.Post,
-                    s_queryUri,
-                    _credentialProvider,
-                    new QuerySpecification { Sql = query });
-                request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json") { CharSet = "utf-8" };
-                if (!string.IsNullOrWhiteSpace(options?.ContinuationToken))
+                async Task<Page<T>> nextPageFunc(string continuationToken, int? pageSizeHint)
                 {
-                    request.Headers.Add(ContinuationTokenHeader, options?.ContinuationToken);
-                }
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Shouldn't this be before the try?

Reply:
This cancellation token check is within the "get next page" func that is called several times over the course of the query. There is a cancellation token check outside the try as well for when the query is being constructed.